src/shogun/lib/any.h

```diff
@@ -172,12 +240,14 @@ namespace shogun
         explicit Any(const T& v) : policy(select_policy<T>()), storage(nullptr)
         {
             policy->set(&storage, &v);
+			m_datatype.set<T>();
```

Comment:
Yeah I thought I could update this after the SGVector part is done because SGObejct requires the serialization of SGVector anyway. But that one is harder than I thought..